design-document.md

```diff
@@ -11,9 +11,98 @@
+                            +-^-------^--+
+                              |       |   
+                          (2) |       | (3)
+                              |       |    
++--------+           +--------v---+   |        +-----------+
+|        |    (1)    |            <---+   (4)  |           |
+| Client +----------->   Leader   +------------> Collector |
+|        |           |            <---+        |           |
++--------+           +--------^---+   |        +-----------+
+                              |       |       
+                          (2) |       | (3)       
+                              |       |        
+                            +-v-------v--+     
+                            |            |     
+                            | Aggregator |
+                            |            |
+                            +------------+
+~~~ 
+
+1. Upload: Clients split inputs into s >= 2 shares, encrypt each share for a different 
+   Aggregator, and send these encrypted shares to the Leader. Clients continue 
+   this process until a "batch" of data is collected.
+2. Verify: Upon receipt of an encrypted share, the Leader sends it to the respective 
+   Aggregator. Each Aggregator decrypts the share, computes a proof from the respective
+   share, sends this proof to the leader. Once the Leader collects all proofs for 
+   the batch, it determines whether or not the data for each entry is correct. 
+   (Details about input validation and how it pertains to system security properties 
+   is in {{CITE}}.)
+3. Aggregate: Assuming the input share is valid, the Leader instructs each Aggregator 
+   to combine aggregate their corresponding input share locally. When complete, each
+   Aggregator sends their aggregated input shares to the Leader, who then combines all
+   aggregates into a final result. 
+4. Collect: The aggregated output is sent to the Collector.
+
+The output of a single batch aggregation reveals little to nothing beyond the value itself.
+
+## Security overview
+
+Prio assumes a powerful adversary with the ability to compromise an unbounded number of 
+clients. In doing so, the adversary can provide malicious (yet truthful) inputs to the aggregation 
+function. Prio also assumes that all but one server operates honestly, where a dishonest
+server does not execute the protocol faithfully as specified. The system also assumes
+that servers communicate over secure and mutually authenticated channels. In practice,
+this can be done by TLS or some other form of application-layer authentication.
+
+In the presence of this adversary, Prio provides two important properties for computing 
+an aggergation function F:
+
+1. Privacy. The adversary learns only the output of F computed over all client inputs, 
+   and nothing else. 
+1. Robustness. The adversary can influence the output of F only by reporting false 
+   (untruthful) data. The output cannot be influenced in any other way.
```

Comment:
The 2017 paper's section 2 discusses _anonymity_ alongside privacy and robustness. However I'm not sure how to draw a clear distinction between anonymity and privacy in this context so maybe it's not worth discussing in this doc.

Reply:
Indeed. I suggest we leave this as-is for now.

Reply:
I would resolve this by adding a refereence to the paper.